File: app/src/main/java/com/remember/app/ui/base/BasePresenter.java
package com.remember.app.ui.base;

import androidx.annotation.NonNull;

import com.arellomobile.mvp.MvpPresenter;
import com.remember.app.NetworkStatus;
import com.remember.app.data.network.ServiceNetwork;

import javax.inject.Inject;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public class BasePresenter<V extends BaseView> extends MvpPresenter<V> {

    @Inject
    protected ServiceNetwork serviceNetwork;

    private CompositeDisposable compositeSubscription = new CompositeDisposable();

    protected void unsubscribeOnDestroy(@NonNull Disposable subscription) {
        compositeSubscription.add(subscription);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        compositeSubscription.clear();
    }

    protected ServiceNetwork getServiceNetwork() {
        return serviceNetwork;
    }

    protected boolean isOffline() {
        if (NetworkStatus.isOffline()) {
            getViewState().onErrorOffline();
            return true;
        }
        return false;
    }

}

File: app/src/main/java/com/remember/app/ui/base/BasePresenter.kt
package com.remember.app.ui.base

import androidx.annotation.NonNull
import com.arellomobile.mvp.MvpPresenter
import com.remember.app.NetworkStatus
import com.remember.app.data.network.ServiceNetwork
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable
import javax.inject.Inject

open class BasePresenter<V : BaseView> : MvpPresenter<V>() {

    @Inject
    protected lateinit var serviceNetwork: ServiceNetwork

    private val compositeSubscription = CompositeDisposable()

    protected fun unsubscribeOnDestroy(@NonNull subscription: Disposable) {
        compositeSubscription.add(subscription)
    }

    override fun onDestroy() {
        super.onDestroy()
        compositeSubscription.clear()
    }

    protected open fun isOffline(): Boolean {
        if (NetworkStatus.isOffline()) {
            viewState.onErrorOffline()
            return true
        }
        return false
    }

}
